src/storage/s3.rs

```diff
@@ -78,11 +91,11 @@ impl S3Backend {
         {
             Ok(_) => Ok(true),
             Err(SdkError::ServiceError(err))
-                if (matches!(err.err(), HeadObjectError::NotFound(_))
-                    || err.raw().status().as_u16() == http::StatusCode::NOT_FOUND.as_u16()) =>
+                if matches!(err.err(), HeadObjectError::NotFound(_)) =>
```

Comment:
Should this check be merged into `err_is_not_found` too?

Reply:
Using just the `E: ProvideErrorMetadata`, I'd assume `e.code() == Some("NotFound")` would cover it (I tried checking what `code` would be returned for this case, but of course there's no easy way to find that in the aws-sdk source).

Reply:
Ah, now I know what you mean. Let me see

Reply:
@Nemo157 that was a nice idea, 

I adapted the logic, while finding the codes in the S3 SDK codebase & validating them with actual S3 requests.